client.go

```diff
@@ -206,8 +212,13 @@ func (k *Kite) NewClient(remoteURL string) *Client {
 		Concurrent:         true,
 		send:               make(chan *message),
 		interrupt:          make(chan error, 1),
+		ctx:                context.TODO(),
```

Comment:
technically todo == background, but seems like we can use background here since the intention is clear for the rest of the flow

possibly this would lead to removal of setContext func

Reply:
Changed to background.